controllers/nodemaintenance_controller.go

```diff
@@ -148,8 +148,11 @@ func (r *NodeMaintenanceReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 		return emptyResult, nil
 	}
 
-	err = initMaintenanceStatus(ctx, nm, drainer, r.Client)
+	err = initMaintenanceStatus(ctx, nm, drainer)
 	if err != nil {
+		if errUpdate := r.Client.Status().Update(ctx, nm); errUpdate != nil {
```

Comment:
why do you update the status here?

Reply:
Previously it was suggested to move the status update outside the initialization function so this function can be used in unit tests. The function fails in the unit test when the update fails.
But we don't need to update it only when there is an initialization error...